File: Data Structures & Algorithms/is-anagram/submission-6.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d1 = {}
        d2 = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            if s[i] in d1:
                d1[s[i]] += 1
            if s[i] not in d1:
                d1[s[i]] = 1
            if t[i] in d2:
                d2[t[i]] += 1
            if t[i] not in d2:
                d2[t[i]] = 1


        if d1 == d2:
            return True
        else: return False
        
        
        
        
        
        
        
        
        
        
        
        
        
        













        
        
        
        # if len(s) != len(t):
        #     return False
        # else:
        #     set(s)
        #     set(t)
        #     if sorted(set(s)) != sorted(set(t)):
        #         return False
        # return True
        

File: Data Structures & Algorithms/two-integer-sum/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */

    //can i saftely assume that the input array is in sorted order

    //we return the inex of the compiment that we saw earlier in the array first since that is the smaller index, then the index of the current value that needs this compliment value to add up to the target
    // we can use an object datastructure to keep track fo the values that we have seen
    twoSum(nums, target) {
        let seen = {}
        for (let i = 0; i < nums.length; i++){
            let compliment = target - nums[i]
            if (compliment in seen){
                return [seen[compliment], i]
            }
            else{
                seen[nums[i]] = i
            }
        }
        
    }
}
